src/cross_module_fn.h

```diff
@@ -139,6 +139,12 @@ typedef struct CrossModuleFunctions
 	PGFunction decompress_chunk;
 	void (*decompress_batches_for_insert)(ChunkInsertState *state, Chunk *chunk,
 										  TupleTableSlot *slot);
+	bool (*decompress_batches_for_update_delete)(Chunk *ch, ScanKeyData *scankeys, int num_scankeys,
+												 Bitmapset *null_columns, List *is_nulls,
+												 bool *chunk_status_changed);
+	ScanKeyData *(*build_update_delete_scankeys)(Chunk *chunk, List *segmentby_columns,
+												 List *segmentby_columns_value, List *is_null_check,
+												 int *num_scankeys, Bitmapset **null_columns);
```

Comment:
why would the ScanKeys ever need to cross the TSL/non-TSL boundary, all the dedcompression should happen localized in the TSL code

Reply:
Move most of code related to constraints handling to compression.c. So above code will no longer be valid.